library/x509_crt.c

```diff
@@ -1184,7 +1184,17 @@ int mbedtls_x509_crt_parse_path( mbedtls_x509_crt *chain, const char *path )
 
     while( ( entry = readdir( dir ) ) != NULL )
     {
-        mbedtls_snprintf( entry_name, sizeof entry_name, "%s/%s", path, entry->d_name );
+        if( strlen(path) + 1 + strlen(entry->d_name) + 1 <= MBEDTLS_X509_MAX_FILE_PATH_LEN )
```

Comment:
The problem is gcc is giving us a warning on *possible* truncation of a string when calling `mbedtls_snprintf`. `mbedtls_snprintf` is safe to call, so there's no need to do this length checking before we call it.

Instead, as @RonEld suggests in the the original bug #1188, and is discussed in the [gcc documentation of the warning](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html), we can just check the return value of `mbedtls_snprintf()` If that is less than the sum of `path`, `entry->d_name` and `/` then we have reason to report an error.